src/__tests__/Document.test.ts

```diff
@@ -856,7 +856,133 @@ describe('validate', () => {
 			});
 			const document = new DocumentSubclass(schema, { data: { prop1: 'foo' } });
 
-			const expected = new Map([['prop1', 'Test error message']]);
+			const expected = new Map([['prop1', ['Test error message']]]);
+			expect(await document.validate()).toEqual(expected);
+		});
+
+		test('return error at property & unravel nested errors if schemaType validation fails', async () => {
+			const passingEmbeddedObjectSchema: SchemaDefinition = {
```

Comment:
These variables which are merely schema definitions are confusing when named `Schema` as they imply that they are schemas. I don't think there's much value in defining them as separate variables -- generally we see Schemas with embedded structures as being defined inline with the `Schema` constructor rather than as separate variables. By inlining it you won't have to worry about confusing variable names.